crates/project_panel/src/project_panel.rs

```diff
@@ -1942,9 +1942,8 @@ impl ProjectPanel {
                 }
                 entry_iter.advance();
             }
-
+            visible_worktree_entries.sort_by(|a, b| a.path.as_ref().cmp(b.path.as_ref()));
```

Comment:
You cannot do that, as `propagate_git_statuses` expects certain file order, otherwise it panics.